fix(qa): cmd/knuckle TUI tests fail in non-interactive ghost QA worktree (no /dev/tty)

## Problem

When `qa-test-pr.sh` runs `just ci` inside a git worktree via `nohup`/`ssh -f` (no PTY allocated), the `cmd/knuckle` integration tests fail:

```
--- FAIL: TestMain_TUINormalMode
    Error: bubbletea: error opening TTY: bubbletea: could not open TTY: open /dev/tty: no such device or address
--- FAIL: TestMain_TUIDemoMode
--- FAIL: TestMain_TUIRebootFnWiring
--- FAIL: TestMain_TUIFlatcarVersionFlag
```

These tests spawn the actual knuckle binary in a subprocess, which tries to open `/dev/tty` to initialize the Bubble Tea TUI.

## Impact

- Every Tier 0 QA run on ghost produces a false CI FAIL for docs/CI-only PRs
- Tests appear to pass when run interactively (SSH with PTY) because results are **cached** from prior interactive runs — masking the issue
- GitHub Actions CI passes correctly (different test runner environment)

## Root Cause

The tests unconditionally open a PTY terminal. When run in a non-interactive subprocess (no controlling terminal), Bubble Tea fails to initialize.

## Fix Options

2. Have the tests pass `--headless` flag to the subprocess so no TTY is required
3. Run the ghost QA script with `ssh -tt` to allocate a PTY (workaround, not a fix)

## Acceptance Criteria

- `just ci` in a non-interactive environment (no `/dev/tty`) exits 0
- `just ci` in an interactive environment still runs all TUI tests

--- a/cmd/knuckle/main_test.go
+++ b/cmd/knuckle/main_test.go
@@ -228,15 +228,33 @@ func TestMain_HeadlessUnwriteableLogFile(t *testing.T) {
 	}
 }
 
+// hasTTY reports whether /dev/tty is accessible in the current process.
+// Used to skip interactive TUI tests in non-interactive environments such as
+// CI runners and ghost QA workers that have no controlling terminal.
+func hasTTY() bool {
+	f, err := os.Open("/dev/tty")
+	if err != nil {
+		return false
+	}
+	_ = f.Close()
+	return true
+}
+
+// skipIfNoTTY calls t.Skip when there is no accessible /dev/tty.
+func skipIfNoTTY(t *testing.T) {
+	t.Helper()
+	if !hasTTY() {
+		t.Skip("skipping TTY test: /dev/tty not accessible (non-interactive environment)")
+	}
+}
+
 // TestMain_TUINormalMode verifies that normal TUI mode starts successfully
 // and follows the full startup path: hardware probe → sysext fetch → channel
 // fetch → tui.Run(). Uses KNUCKLE_TEST_TUI_AUTO_QUIT to exit cleanly.
 // Covers: log file setup, channel validation, runner setup, prober/bakery/
 // installer wiring, wizard creation, probe/fetch calls, tui.Run() invocation.
 func TestMain_TUINormalMode(t *testing.T) {
-	if os.Getenv("CI") != "" {
-		t.Skip("skipping TTY test in CI — no /dev/tty available")
-	}
+	skipIfNoTTY(t)
 	logFile := t.TempDir() + "/knuckle.log"
 	cmd := helperCmdWithTimeout(t, 15*time.Second,
 		"--channel=stable", "--dry-run", "--log-file="+logFile)
@@ -256,9 +274,7 @@ func TestMain_TUINormalMode(t *testing.T) {
 // real hardware/network calls. Covers the demo mode branch, mock wiring, and
 // pre-populated wizard state.
 func TestMain_TUIDemoMode(t *testing.T) {
-	if os.Getenv("CI") != "" {
-		t.Skip("skipping TTY test in CI — no /dev/tty available")
-	}
+	skipIfNoTTY(t)
 	logFile := t.TempDir() + "/knuckle-demo.log"
 	cmd := helperCmdWithTimeout(t, 15*time.Second,
 		"--demo", "--log-file="+logFile)
@@ -298,9 +314,7 @@ func TestMain_TUILogFileError(t *testing.T) {
 // mode. The auto-quit mechanism prevents the TUI from blocking.
 // Covers: rebootFn setup for non-dry-run mode (lines 147-152 in main.go).
 func TestMain_TUIRebootFnWiring(t *testing.T) {
-	if os.Getenv("CI") != "" {
-		t.Skip("skipping TTY test in CI — no /dev/tty available")
-	}
+	skipIfNoTTY(t)
 	logFile := t.TempDir() + "/knuckle-reboot.log"
 	// Start TUI without --dry-run (rebootFn will be non-nil)
 	// but use auto-quit to prevent blocking. Since we quit before install
@@ -376,9 +390,7 @@ func TestRunHeadless_DryRunSuccess(t *testing.T) {
 // TestMain_TUIFlatcarVersionFlag verifies that the --flatcar-version flag
 // is correctly passed to the wizard state.
 func TestMain_TUIFlatcarVersionFlag(t *testing.T) {
-	if os.Getenv("CI") != "" {
-		t.Skip("skipping TTY test in CI — no /dev/tty available")
-	}
+	skipIfNoTTY(t)
 	logFile := t.TempDir() + "/knuckle-version.log"
 	cmd := helperCmdWithTimeout(t, 15*time.Second,
 		"--channel=stable", "--dry-run", "--flatcar-version=3510.2.8", "--log-file="+logFile)

